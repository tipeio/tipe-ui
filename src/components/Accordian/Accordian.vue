<template>
  <div class="content">
    <div 
      class="row" 
      @click="onClick">
      <p class="title" >{{ text.toUpperCase() }}</p>
      <tipe-icon
        :class="{open}"
        icon="arrow-down"
        class="icon"
      />
    </div>
    <slot v-if="open"/>
  </div>
</template>

<script>
import vueTypes from 'vue-types'
import TipeIcon from '../Icon'

export default {
  name: 'Accordian',
  components: {
    TipeIcon
  },
  props: {
    text: vueTypes.string
  },
  data: function() {
    return {
      open: false
    }
  },
  methods: {
    onClick() {
      this.open = !this.open
    }
  }
}
</script>

<style scoped lang="postcss">
.content {
  display: flex;
  align-items: flex-start;
  justify-content: center;
  flex-direction: column;
  width: 199px;

  & .row {
    display: flex;
    align-items: center;
    justify-content: space-between;
    height: 2.125rem;
    width: 100%;
    border-bottom: 1px solid #eeeeee;
    margin-bottom: 4px;

    & .icon {
      display: flex;
      flex: 0 0 2rem;
      transition: transform 250ms ease-in-out;

      &.open {
        transform: rotate(-180deg);
      }
    }
    & p {
      margin: 0;
      font-size: 0.6875rem;
      color: #dce0e5;
    }
  }
}
</style>
