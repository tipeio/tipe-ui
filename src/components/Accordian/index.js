export { default } from './Accordian.vue'
