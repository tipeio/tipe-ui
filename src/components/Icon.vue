<template>
  <div
    :class="size"
    :data-tipe-ui="$options.name"
  />
</template>

<script>
import vueTypes from 'vue-types'

const sizes = ['small', 'medium', 'large']

export default {
  name: 'TipeIcon',
  props: {
    size: vueTypes.oneOf(sizes).def('medium')
  }
}
</script>

<style scoped>
[data-tipe-ui='TipeIcon'] {
  background-color: red;
}

[data-tipe-ui='TipeIcon'].small {
  height: 1rem;
  width: 1rem;
}

[data-tipe-ui='TipeIcon'].medium {
  height: 2rem;
  width: 2rem;
}

[data-tipe-ui='TipeIcon'].large {
  height: 3rem;
  width: 3rem;
}
</style>
