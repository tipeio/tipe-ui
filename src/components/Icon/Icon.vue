<template>
  <object
    :data="src"
    type="image/svg+xml">
    <img :src="src" >
  </object>
</template>

<script>
import vueTypes from 'vue-types'

export default {
  name: 'TipeIcon',
  props: {
    icon: vueTypes.string
  },
  computed: {
    src() {
      return require(`../../icons/${this.icon}.svg`)
    }
  }
}
</script>
