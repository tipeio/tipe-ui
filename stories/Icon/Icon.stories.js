import { storiesOf } from '@storybook/vue'

import TipeIcon from '../../src/components/Icon'

const style = () => ({
  height: '1rem',
  width: '1rem'
})

const icons = [
  'check',
  'document',
  'folder',
  'key',
  'number',
  'share',
  'switch',
  'user',
  'color',
  'edit',
  'gear',
  'link',
  'pin',
  'sign-out',
  'text',
  'view',
  'copy',
  'flow',
  'grid',
  'list',
  'search',
  'star-active',
  'trash',
  'wrench',
  'credit-card',
  'folder-open',
  'group',
  'markdown',
  'media',
  'send',
  'star',
  'user-circle',
  'options',
  'plus',
  'cross',
  'arrow-down',
  'arrow-up',
  'arrow-right',
  'arrow-left'
]

storiesOf('Icon', module).add('icons', () => ({
  components: { TipeIcon },
  computed: { style },
  template:
    '<div>' +
    icons.reduce((template, icon) => {
      return (
        template +
        `<tipe-icon :style="style" icon="${icon}" color="rgb(255,0,0)" />`
      )
    }, '') +
    '</div>'
}))
