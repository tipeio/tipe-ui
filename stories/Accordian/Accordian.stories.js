import { storiesOf } from '@storybook/vue'

import Accordian from '../../src/components/Accordian'
import { ActivityItem } from '../../src/components/AccountSettings/RecentActivity'

const item = () => ({ title: 'Sam Smith update API key', time: '1 hour ago' })
const style = () => ({ display: 'grid', 'grid-row-gap': '1rem' })

storiesOf('Accordian', module)
  .add('accordian', () => ({
    components: { Accordian, ActivityItem },
    computed: { item },
    template:
      '<accordian text="Recent Activity"><activity-item :item="item"/></accordian>'
  }))
  .add('two accordians', () => ({
    components: { Accordian, ActivityItem },
    computed: { item, style },
    template:
      '<div :style="style"><accordian text="Recent Activity"><activity-item :item="item"/></accordian><accordian text="Recent Activity"><activity-item :item="item"/></accordian></div>'
  }))
