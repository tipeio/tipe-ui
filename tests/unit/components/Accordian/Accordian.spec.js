import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import Accordian from '@/components/Accordian'
import Icon from '@/components/Icon'

describe('Accordian.vue', () => {
  it('matches previous snapshot', async () => {
    const wrapper = shallowMount(Accordian, {
      propsData: {
        text: 'My Stuff'
      }
    })
    await expect(createRenderer().renderToString(wrapper.vm)).toMatchSnapshot()
  })
  it('should have the content classs', () => {
    const wrapper = shallowMount(Accordian, {
      propsData: {
        text: 'My Stuff'
      }
    })
    expect(wrapper.classes()).toContain('content')
  })
  it('should have the correct uppercased title', () => {
    const wrapper = shallowMount(Accordian, {
      propsData: {
        text: 'My Stuff'
      }
    })
    expect(wrapper.find('.title').text()).toBe('MY STUFF')
  })
  it('should render the icon component', () => {
    const wrapper = shallowMount(Accordian, {
      propsData: {
        text: 'My Stuff'
      }
    })
    expect(wrapper.contains(Icon)).toBe(true)
  })
  it('should not render the slotOpen class by default', () => {
    const wrapper = shallowMount(Accordian, {
      propsData: {
        text: 'My Stuff'
      }
    })
    expect(wrapper.find('.slotOpen').exists()).toBe(false)
  })
  it('should show slot onClick', () => {
    const wrapper = shallowMount(Accordian, {
      propsData: { text: 'Settings' }
    })
    wrapper.trigger('click')
    expect(wrapper.find('.slotOpen').exists()).toBe(true)
  })
  it('should toggle the slot', () => {
    const wrapper = shallowMount(Accordian, {
      propsData: { text: 'Settings' }
    })
    wrapper.trigger('click')
    expect(wrapper.find('.slotOpen').exists()).toBe(true)
    wrapper.trigger('click')
    expect(wrapper.find('.slotOpen').exists()).toBe(false)
  })
})
