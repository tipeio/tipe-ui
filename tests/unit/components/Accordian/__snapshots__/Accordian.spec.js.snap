// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian.vue matches previous snapshot 1`] = `Promise {}`;
